### DISCUSSION TITLE
@persist directive breaks when persisting a Livewire component

### DISCUSSION THREAD

**Discussion by WyattCast44** [livewire/livewire#6791] [7. Bugs (archived)]
### Livewire version

3.0.3

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. Create a livewire component
2. Wrap the `@livewire('name')` directive in a `@persist('name')` directive
```blade
@persist('comp-name')
        @livewire('comp-filename')
@endpersist
```
4. Navigate the application with links marked with the `wire:navigate` attribute  
5. Check the console for errors
6. Note the following error: `livewire.esm.js:6441 Uncaught Component already initialized`
7. Check the source file, find this class constructor 
```js
var Component = class {
  constructor(el) {
    if (el.__livewire)
      throw "Component already initialized"; // HERE is where the error is thrown
    el.__livewire = this;
    this.el = el;
    this.id = el.getAttribute("wire:id");
    this.__livewireId = this.id;
    this.snapshotEncoded = el.getAttribute("wire:snapshot");
    this.snapshot = JSON.parse(this.snapshotEncoded);
    if (!this.snapshot) {
      throw `Snapshot missing on Livewire component with id: ` + this.id;
    }
    this.name = this.snapshot.memo.name;
    this.effects = JSON.parse(el.getAttribute("wire:effects"));
    this.originalEffects = deepClone(this.effects);
    this.canonical = extractData(deepClone(this.snapshot.data));
    this.ephemeral = extractData(deepClone(this.snapshot.data));
    this.reactive = Alpine.reactive(this.ephemeral);
    this.$wire = generateWireObject(this, this.reactive);
    this.cleanups = [];
    processEffects(this, this.effects);
  }
// more stuff
}
```

### Additional information

1. I cant find any disclaimer in the Livewire documentation that a Livewire component cannot be persisted just like any other element. 
2. I have no idea if this is possible from a technical perspective 😅
3. It would be super cool if it was 🔥

My use case is: I am trying to persist a Livewire component that uploads files. Ideally, I'd like the user to open a slide-over panel, drag and drop some files. Those files start uploading using the JS API methods. The user is free to navigate the rest of the application while the files upload in the background.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by loxi5** [comment]
Any progress on this?

**Comment by gdebrauwer** [comment]
Maybe related to https://github.com/livewire/livewire/pull/7881 as that triggers the same "Uncaught Component already initialized" error

**Comment by gdebrauwer** [comment]
@WyattCast44 I attempted to create a failing test (https://github.com/gdebrauwer/livewire/commit/92cd46e83636079c2a7e43f561d758b64a418344), but the test did not fail 😅 Are you doing something special?

**Comment by DaniilSkLi** [comment]
I have the similar issue with `@persist`. The app layout uses the `<livewire:notifications />` component.  
The contents of this component:
```blade
@persist('notifications')
    <div class="notifications">
        123
    </div>
@endpersist
```

When I navigate through pages everything is fine, but as soon as I use the back arrow in the browser, I get the error `Uncaught Component already initialized`. If I remove `@persist` - the error does not appear.

Livewire 3.5.4  
PHP 8.2.20

<details><summary>App layout</summary>
<p>

```blade
<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <link href="{{ asset('favicon.png') }}" rel="icon">

        <title>{{ isset($title) ? $title . ' | ' . env('APP_NAME') : env('APP_NAME') }}</title>

        @vite(['resources/css/app.styl'])
    </head>
    <body id="app-layout">
        @section('main')
            {{ $slot }}
        @show

        <livewire:notifications />

        @vite(['resources/js/app.js'])
    </body>
</html>
```

</p>
</details>

**Comment by joshhanley** [comment]
@WyattCast44 thanks for reporting! We're in the process of investigating all `wire:navigate` issues and I have tested out your scenario. I believe this has been fixed now in PR #9339. It will be available in the next release. One thing to note though, is that the component will "mount" again on the backend, but not the front end. The solution to this is to mark the component as `lazy`. See here for more details https://github.com/livewire/livewire/discussions/9177#discussioncomment-13416909
